### PR TITLE
Use Tag in IdentifierContainer

### PR DESCRIPTION
**Pull request by denysoblohin-okta**
## Description:

- Updated Odyssey to 1.9.16
- Added `translate="no"` for `Callout` and `Tag`. For now it's being rendered as `translate="yes"` ([example in snapshots](https://github.com/okta/okta-signin-widget/pull/3487/files#diff-55b815d63b27f3d8b8ee531db22a4eca16ac367376f011d84340191275916e2aR34)) due to bug in preact https://github.com/preactjs/preact/issues/3799 until https://github.com/preactjs/preact/pull/3800 is merged
- Modified English Leaks test to support [both `.no-translate, [translate="no"]`](https://github.com/okta/okta-signin-widget/pull/3487/files#diff-20ecefc4d164e1f22564fa444b02083080727de4a7c205be486bfbacbbbd050dR11)   (until all `no-translate` classes are removed and issue with `translate="yes"` is fixed). 
No-translation content should be get with [using `innerText`](https://github.com/okta/okta-signin-widget/pull/3487/files#diff-20ecefc4d164e1f22564fa444b02083080727de4a7c205be486bfbacbbbd050dR174) instead of `textContent` - same as for [getting all visible text on page](https://github.com/okta/okta-signin-widget/pull/3487/files#diff-20ecefc4d164e1f22564fa444b02083080727de4a7c205be486bfbacbbbd050dR165), to to avoid including hidden elements

## PR Checklist

- [x] Have you verified the basic functionality for this change?
- [ ] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [x] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [x] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- [OKTA-674223](https://oktainc.atlassian.net/browse/OKTA-674223)

### Screenshot/Video:

Before:
Screenshot 2023-12-18 at 17 33 51

After:
Screenshot 2023-12-18 at 17 29 42

### Downstream Monolith Build:

https://bacon-go.aue1e.saasure.net/commits?artifact=monolith&branch=d16t-okta-signin-widget-3790c27-658356cc&page=1&pageSize=6&sha=79b5070bbf78117c42ab596fe872692da49173da&tab=main

### CODE REVIEW

File: src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx
Comment (lesterchoi-okta) [inline src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx:42]: Shouldn't these use design tokens?

File: src/v3/test/integration/__snapshots__/authenticator-email-verification-data.test.tsx.snap
Comment (lesterchoi-okta) [inline src/v3/test/integration/__snapshots__/authenticator-email-verification-data.test.tsx.snap:119]: Are these attributes ok to remove? `data-se`, `aria-label`, `title` seem important

**Review by lesterchoi-okta — COMMENTED** [review]
Question about missing attributes in snapshots after change

File: src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx
Comment (denysoblohin-okta) [reply src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx:42]: There is the dedicated ticket to review icon sizing: [OKTA-671459](https://oktainc.atlassian.net/browse/OKTA-671459)

File: src/v3/test/integration/__snapshots__/authenticator-email-verification-data.test.tsx.snap
Comment (glenfannin-okta) [reply src/v3/test/integration/__snapshots__/authenticator-email-verification-data.test.tsx.snap:119]: I believe there are selenium tests within the monolith that uses this data-se attribute (we should probably be periodically running the downstreams against this topic branch to avoid a delay once this is completed. 

Also, re: the title, this was used because if the user ID is extremely long, we have css that truncates it and adds an ellipsis, the title was for visual users that allows them to hover the container to see the full user ID. For that reason I think we may need to keep this unless design says otherwise (it was originally a directive from UX to add this).

File: src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx
Comment (glenfannin-okta) [inline src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx:50]: Was there a reason this css was removed (specifically the textOverflow, whitespace and overflow properties)? This is to truncate long user IDs to avoid the container from taking up multiple rows (this was a UX decision).

File: src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx
Comment (trevoring-okta) [reply src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx:26]: We've been having ongoing discussions with Odyssey about how we can move `no-translate` to instead use the `translate="no"` global HTML attribute (this is what I talked to you about for the checkbox component, @glenfannin-okta). I have an open PR to get this merged into Odyssey right now, which would unblock this for the identifier container. Maybe we should wait until that gets merged and then we can bump to latest version of Odyssey

File: src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx
Comment (trevoring-okta) [reply src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx:50]: I think this is because Odyssey basically does not expose `sx` on any non-Box component. I've been anticipating that many of our components have built-in one-offs from product/UX like this one that won't play well with Odyssey. I'm not sure if there's another way to achieve this fix so if it's deemed critical we might have to flag this component as one that cannot be converted to use Odyssey's `Tag`.

File: src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx
Comment (glenfannin-okta) [reply src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx:26]: Ahh, that makes sense, this actually reminds me, if you guys change how things are tagged as "no to be translated" do you also plan to update the corresponding Selenium tests in the monolith that looks for that class on the SIW elements? I just want to make sure we have accounted for those tests as well.

File: src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx
Comment (glenfannin-okta) [reply src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx:50]: Ah I see, could we just wrap the text in a box component (span) and apply this styling? Would that cause any issues?

File: src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx
Comment (trevoring-okta) [reply src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx:26]: Yeah I believe we had this discussion sometime when you were on PTO but @lesterchoi-okta was doing an investigation into the selectors that are used in monolith and whether it will be feasible to convert all of them over to semantic selectors or `data-se`. I'll start a wiki right now to keep track of selectors that we know we've broken or removed as part of the migration

File: src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx
Comment (trevoring-okta) [reply src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx:50]: Yeah I just tested this and I think we can do this. @denysoblohin-okta it seems to me like we can move this text overflow styling to the root `Box` element and also remove some of the styles on that root element because they're not needed now that we have `Tag` styling. Note: we'll have to continue using the normal MUI `Box` that we import from `@okta/odyssey-react-mui-legacy` because OD 1.x `Box` doesn't support `title`

I'm thinking something like this:
```
<Box
  title={identifier}
  sx={{
    textOverflow: 'ellipsis',
    whiteSpace: 'nowrap',
    overflow: 'hidden',
  }}
  data-se="identifier-container"
>
  <Tag
    icon={(
      <Box
        sx={{
          width: '16px',
          height: '16px',
        }}
      >
        <UserIcon
          titleAccess="User"
          sx={{
            color: tokens.PalettePrimaryMain,
            width: '16px',
            height: '16px',
          }}
        />
      </Box>
    )}
    label={identifier}
    testId="identifier"
  />
</Box>
```

File: src/v3/test/integration/__snapshots__/authenticator-email-verification-data.test.tsx.snap
Comment (denysoblohin-okta) [reply src/v3/test/integration/__snapshots__/authenticator-email-verification-data.test.tsx.snap:119]: As for `title` attribute, I can add it to the outer `Box` instead of `Tag` (which doesn't support it). Then it would show tooltip as before.

As for `data-se="identifier"`, it's applied to outer `div` (line 92)
I hope monolith selenium tests don't use `.identifier` class.
And I hope it apply `innerText` instead of `textContent` against element with `data-se="identifier"`. Otherwise we can remove `titleAccess="User"` from `<UserIcon>` to delete `<title>User</title>` inside icon

File: src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx
Comment (denysoblohin-okta) [reply src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx:26]: As a quick fix for enleaks tests in SIW, I've modified spec by adding `.identifier-container span` in addition to `.no-translate` as selector for text without translation
https://github.com/okta/okta-signin-widget/pull/3487/commits/60e0ccb46256f738c3fc1acbfc4fda9f093b1149#diff-20ecefc4d164e1f22564fa444b02083080727de4a7c205be486bfbacbbbd050d
But it's bad workaround, should remove it.

A better way would be to add class `no-translate` to the outer `Box` (the one with `identifier-container` class). But the problem is that its `.textContent` would include `User` from `<title>User</title>` inside icon.
Any reason why `.textContent` is used instead of `.innerText` ?
https://github.com/okta/okta-signin-widget/blob/b67d788ed46fad0d11c981895969243103fed752/test/testcafe/spec-en-leaks/EnglishLeaks_spec.js#L173

Comment (trevoring-okta) [reply src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx:26]: I'm not 100% sure how our English leak tests work but I'm assuming we use `.textContent` to also exclude non-visible text that appears in the DOM but not on the page because I believe otherwise our English leak tests will flag non-visible text like `<title>User</title>` as a leak?

TLDR: I think you should be safe to move the `.no-translate` to the outer `Box` for now and then we can remove it once https://github.com/okta/odyssey/pull/2074 is merged in to Odyssey

Comment (denysoblohin-okta) [reply src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx:26]: Actually `.textContent` **includes** non-visible text (like content of `title` tag) while `.innerText` does not include.

So if we mix `.textContent`  and `.inerText`
https://github.com/okta/okta-signin-widget/blob/b67d788ed46fad0d11c981895969243103fed752/test/testcafe/spec-en-leaks/EnglishLeaks_spec.js#L164
https://github.com/okta/okta-signin-widget/blob/b67d788ed46fad0d11c981895969243103fed752/test/testcafe/spec-en-leaks/EnglishLeaks_spec.js#L173
it cause problems with hidden text

File: test/testcafe/spec-en-leaks/EnglishLeaks_spec.js
Comment (denysoblohin-okta) [inline test/testcafe/spec-en-leaks/EnglishLeaks_spec.js:13]: Should be removed after merging of https://github.com/okta/odyssey/pull/2074/files#diff-ba66395dd166ceaa1fb9be3e394dc3b399a088fdf32cbe51311b64f83f99ab8dR61

**Comment by denysoblohin-okta** [comment]
Checked monolith downstream tests. 
Looks like this PR hasn't introduced new failures.
Downstream for this PR: https://bacon-go.aue1e.saasure.net/commits?artifact=monolith&branch=d16t-okta-signin-widget-3790c27-658356cc&page=1&pageSize=6&sha=79b5070bbf78117c42ab596fe872692da49173da&tab=main
Downstream for base commit from `odyssey-1.x` branch: https://bacon-go.aue1e.saasure.net/commits?artifact=monolith&branch=d16t-okta-signin-widget-56e48ef-6583ff9f&page=1&pageSize=6&sha=3ecf5a9647cb66857cc850b45c3f0cf7ff43eff2&tab=main

File: src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx
Comment (trevoring-okta) [inline src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx:34]: Can we switch this to `translate="no"` since that should be possible with a MUI `Box` and then remove the `className`

**Review by trevoring-okta — APPROVED** [review]
LGTM besides one question about removing `.no-translate` from MUI `Box` in `IdentifierContainer` component

File: src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx
Comment (trevoring-okta) [reply src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx:34]: That's fair, so you already tested and removing this `.no-translate` will break english leak CI? I might have a potential fix for the Preact issue coming

Comment (denysoblohin-okta) [reply src/v3/src/components/IdentifierContainer/IdentifierContainer.tsx:34]: Actually for now english leaks tests are being run in CI only for v2, not for v3 
I see test failures when running manually `yarn test:testcafe-run-en-leaks` (with `yarn workspace v3 dev`)
So I think it's fine to remove classes `no-translate` already and probably enable english leaks tests for v3 in the future

**Pull request merged**
